src/Api/Commands/SplitToDiscussionHandler.php

```diff
@@ -0,0 +1,195 @@
+
+        $command->actor->assertCan('split', $startPost->discussion);
+
+        /** @var Discussion $originalDiscussion */
+        $originalDiscussion = $startPost->discussion;
+
+        $discussion = Discussion::find($command->discussion_id);
+
+        // update all posts that are split.
+        $affectedPosts = $this->assignPostsToDiscussion(
+            $originalDiscussion,
+            $discussion,
+            $startPost->number,
+            $command->end_post_number
+        );
+
+        $this->renumberDiscussion($originalDiscussion);
+        $originalDiscussion = $this->refreshDiscussion($originalDiscussion);
+
+        $this->renumberDiscussion($discussion);
+        $discussion = $this->refreshDiscussion($discussion);
+
+        $this->events->dispatch(
+            new DiscussionWasSplit($command->actor, $affectedPosts, $originalDiscussion, $discussion)
+        );
+
+        return $discussion;
+    }
+
+    /**
+     * Assign the specific range to a new Discussion without resetting post numbers.
+     *
+     * @param Discussion $originalDiscussion
+     * @param Discussion $discussion
+     * @param            $startPostNumber
+     * @param            $endPostNumber
+     *
+     * @return Collection
+     */
+    protected function assignPostsToDiscussion(
+        Discussion $originalDiscussion,
+        Discussion $discussion,
+        $startPostNumber,
+        $endPostNumber
+    ): Collection {
+        $lastPostNumber = $this->posts
+            ->query()
+            ->where('discussion_id', $discussion->id)
+            ->max('number');
+
+        $this->posts
+            ->query()
+            ->where('discussion_id', $originalDiscussion->id)
+            ->whereBetween('number', [$startPostNumber, $endPostNumber])
+            ->each(function (Post $post) use (&$lastPostNumber, $discussion) {
+                $lastPostNumber++;
+                $post->discussion_id = $discussion->id;
+                $post->number = $lastPostNumber;
+                $post->save();
+            });
+
+        // Update relationship posts on new discussion.
+        $discussion->load('posts');
+
+        $discussion->post_number_index = $lastPostNumber;
+        $discussion->save();
+
+        return $discussion->posts;
+    }
+
+    /**
+     * Re-assign numbers starting from one to a discussion.
+     *
+     * @param Discussion $discussion
+     */
+    protected function renumberDiscussion(Discussion $discussion)
+    {
+        $db = resolve('db');
+
+        $db->statement('SET @maxNumber = (SELECT MAX(number) FROM posts WHERE discussion_id = ?);', [$discussion->id]);
+        $db->statement('SET @rank = @maxNumber;');
+        $db->statement('UPDATE posts SET number=@rank:=@rank+1 WHERE discussion_id = ? ORDER BY created_at;', [$discussion->id]);
+        $db->statement('UPDATE posts SET number=number-@maxNumber WHERE discussion_id = ?;', [$discussion->id]);
```

Comment:
This raw SQL will not work with table prefixes. I'm also not good at SQL so I'm not going to comment on the statements themselves, performance, etc.